### ISSUE TITLE
File upload size limit?

### ISSUE THREAD

**Issue by TdoubleG** [slimphp/Slim#2319]
Hello,

is there any Limit when uploading a file?

I want to upload an image and every image lower then 2MB works. Once the file is bigger then 2 MB I get the following error:

```
Warning: fopen(): Filename cannot be empty in 
<server url>/slim/vendor/slim/slim/Slim/Http/UploadedFile.php on line 189
Slim Application Error
The application could not run because of the following error:

Details
Type: InvalidArgumentException
Message: Slim\Http\Stream::attach argument must be a valid PHP resource
File: <server url>/slim/vendor/slim/slim/Slim/Http/Stream.php
Line: 149
```

Is it possible to change the sizings?

**Comment by tflight** [comment]
That is typically a setting within your webserver (Apache, Nginx, etc) not within Slim. You can typically have a look in `phpinfo()` and see what it says under `upload_max_filesize`.

**Comment by TdoubleG** [comment]
Thank you this was the problem! 
Could it be possible to show a correct error?

**Comment by geggleto** [comment]
its not possible to show a better error; :(

**Comment by tflight** [comment]
Depending on the type of users you have, I will often check the `upload_max_filesize` setting and present it to the user before they upload. That avoids the issue altogether, but isn't possible in all scenarios.

**Comment by akrabat** [comment]
I would expect `getError()` to be non-zero.

**Comment by pachico** [comment]
tl;dr.
Actually, you could show a better error but it's up to you, and not to Slim, to implement a custom error handler (http://php.net/manual/es/function.set-error-handler.php).
It is also a good way to have all your logs centralised in the same way/place, otherwise you will get your custom logs written where you specified (in case of the typical file error Monolog) and the rest of your logs (like this one) written where your nginx/apache configs are set to write to.
In addition, this would allow you to throw exceptions on notices and warnings, which is a good thing if you don't want to have unexpected behaviours (like automatic casting non-existing array key values to null, which might lead to nightmare debugging nights).

Implementing it shouldn't take you more than a couple of hours, tests included :)

**Comment by TdoubleG** [comment]
Thank you guys for the help. I close the ticket :)
